nominal/core/unit.py

```diff
@@ -28,3 +38,57 @@ def _from_conjure(cls, api_unit: api.Unit | scout_units_api.Unit | scout_run_api
             return cls(name="", symbol=api_unit)
         name = "" if api_unit.name is None else api_unit.name
         return cls(name=name, symbol=api_unit.symbol)
+
+
+UnitLike: TypeAlias = Union[Unit, str, None]
+UnitMapping: TypeAlias = Mapping[str, UnitLike]
+
+
+def _unit_symbol_from_unit_like(unit: UnitLike) -> str | None:
+    if unit is None:
+        return None
+    elif isinstance(unit, Unit):
+        return unit.symbol
+    else:
+        return unit
+
+
+def _available_units(auth_header: str, client: scout.UnitsService) -> Sequence[Unit]:
+    """Retrieve the list of all allowable units within Nominal"""
+    response = client.get_all_units(auth_header)
+    return [Unit._from_conjure(unit) for units in response.units_by_property.values() for unit in units]
+
+
+def _build_unit_update(unit: UnitLike) -> timeseries_logicalseries_api.UnitUpdate:
+    """Helper function for building a UnitUpdate enum from a potentially null unit symbol."""
+    unit = _unit_symbol_from_unit_like(unit)
+    if unit is None:
+        return timeseries_logicalseries_api.UnitUpdate(clear_unit=api.Empty())
+    else:
+        return timeseries_logicalseries_api.UnitUpdate(unit=unit)
+
+
+def _warn_on_invalid_units(unit_map: UnitMapping, unit_service: scout.UnitsService, auth_header: str) -> None:
+    # Normalize unit map to refer to channel names and unit symbols
+    channels_to_units = {channel: _unit_symbol_from_unit_like(unit) for channel, unit in unit_map.items()}
+
+    resolved_units = unit_service.get_batch_units(
+        auth_header, [unit_symbol for unit_symbol in set(channels_to_units.values()) if unit_symbol is not None]
+    )
+
+    # Get set of all provided invalid units
+    invalid_units = set(
+        [unit_symbol for unit_symbol in channels_to_units.values() if unit_symbol not in resolved_units]
+    )
```

Comment:
if you store the `set(channels_to_units.values())` above you can just do a set difference here